src/server/node_func.c

```diff
@@ -211,6 +211,36 @@ char *nodename;
 }
 
 
+/**
+ * @brief
+ * 		find_nodebyaddr() - find a node host by its addr
+ * @param[in]	addr	- addr being searched
+ *
+ * @return	pbsnode
+ * @retval	NULL	- failure
+ */
+
+struct pbsnode	  *find_nodebyaddr(addr)
+pbs_net_t addr;
+{
+	int i, j;
+	mom_svrinfo_t *psvrmom;
+
+	for (i=0; i<svr_totnodes; i++) {
+		if (pbsndlist[i]->nd_state & INUSE_DELETED)
```

Comment:
Shouldn't this check be moved to the caller? a find function can just find the node by matching the address and caller can decide whether node is in acceptable state.
If you think otherwise then please update the description of the function header to say that it will ignore nodes in state "INUSE_DELETED"